samples/nrf9160/modem_trace_flash/src/main.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/uart.h>
+#include <modem/lte_lc.h>
+#include <modem/nrf_modem_lib.h>
+#include <modem/nrf_modem_lib_trace.h>
+#include <nrf_modem.h>
+#include <nrf_modem_at.h>
+#include <dk_buttons_and_leds.h>
+
+#include "trace_flash.h"
+
+LOG_MODULE_REGISTER(modem_trace_flash_sample, CONFIG_MODEM_TRACE_FLASH_SAMPLE_LOG_LEVEL);
+
+#define UART1_DT_NODE DT_NODELABEL(uart1)
+
+static const struct device *const uart_dev = DEVICE_DT_GET(UART1_DT_NODE);
+
+/* define callback */
+LTE_LC_ON_CFUN(cfun_hook, on_cfun, NULL);
+
+/* callback implementation */
+static void on_cfun(enum lte_lc_func_mode mode, void *context)
+{
+	LOG_INF("LTE mode changed to %d\n", mode);
+}
+
+static void print_uart1(char *buf, int len)
+{
+	if (!device_is_ready(uart_dev)) {
+		LOG_ERR("uart1 device not found!");
```

Comment:
```suggestion
		LOG_ERR("uart1 device not ready!");
```

Reply:
It's typically not ready because the device node is not found (not enabled or typo). This error is what they use in the uart/echo_bot sample upstream, and I think it helps the user to better understand what to look for. I'm not 100% it can not be ready for other reasons, so can change it to `found/ready` just in case.